### DISCUSSION TITLE
Client side routing 404

### DISCUSSION THREAD

**Discussion by fluteboi** [vikejs/vike#1187] [Discussion]
Hi. I am having some issue with client side routing.

SSR is fine. All page are render without issue when rendered from server, but when I click on any links to do client side routing, I get the following message in the console

`[vike] throw render(404, "Page not found") intercepted while rendering /some-url/`

The render function in _default.page.client.js was not called before the throw render(), nor the onBeforeRender. I cannot seem to figure out what is triggering the 404. The network call to index.pageContext.json returned with the following

`{ abortStatusCode: 404, is404: false }`

Any suggestion on where to look is much appreciated. Thank you!

**Comment by brillout** [comment]
Do you have a [`throw render()`](https://vike.dev/render) call in your code base?

**Reply by fluteboi** [reply]
I do have one in onBeforeRender, but I added a console log just before the `throw render()` and I can see that my `throw render()` was not the one triggering the 404.

**Reply by brillout** [reply]
It's most likely a user land issue.

Dig more (https://brillout.github.io/rules/#try) and if you still believe it to be a Vike issue then create a minimal reproduction with steps showcasing what you expect and what happens instead.

**Reply by fluteboi** [reply]
You are correct. It was an issue on my end. I was able to fix it and everything is good now. Thanks for the quick response.

**Reply by brillout** [reply]
👍 I'm glad you sorted it out. Btw. would your company be up for [sponsoring](https://github.com/sponsors/brillout)?

**Reply by fluteboi** [reply]
Let me speak to the team. Thanks again for the help!

**Reply by brillout** [reply]
Thanks! Let me know how it goes, also if it's a no.
